org.idempiere.test/src/org/idempiere/test/adwindow/GridWindowTest.java

```diff
@@ -0,0 +1,280 @@
+		assertTrue(gridTabVOsCache.getHit() > gridTabVOsCacheHit, "GridTabVOs cache hit doesn't increase as expected");		
+		gridWindowVOCacheHit = gridWindowVOCache.getHit();
+		gridTabVOsCacheHit = gridTabVOsCache.getHit();
+		
+		//test GridWindowVO.create call with not advanced role
+		Env.setContext(Env.getCtx(), Env.AD_ROLE_ID, DictionaryIDs.AD_Role.GARDEN_WORLD_ADMIN_NOT_ADVANCED.id);		
+		windowVO = GridWindowVO.create(Env.getCtx(), 1, SystemIDs.WINDOW_TEST);
+		fields = windowVO.Tabs.get(0).getFields();
+		optional = fields.stream().filter(e -> e.AD_Field_ID == testRecordIdFieldId).findFirst();
+		assertFalse(optional.isPresent(), "Record ID field of Test Window found for not advanced role");
+		
+		assertTrue(gridWindowVOCache.getHit() > gridWindowVOCacheHit, "GridWindowVO cache hit doesn't increase as expected");
+		//gridTabVOs cache for advanced role will not get hit for GridWindowVO.create call with not-advanced role
+		assertTrue(gridTabVOsCache.getHit() == gridTabVOsCacheHit, "Unexpected increase of GridTabVOs cache hit");		
+		gridWindowVOCacheHit = gridWindowVOCache.getHit();
+		gridTabVOsCacheHit = gridTabVOsCache.getHit();
+		
+		//test GridWindowVO.get call with advanced role
+		Env.setContext(Env.getCtx(), Env.AD_ROLE_ID, DictionaryIDs.AD_Role.GARDEN_WORLD_ADMIN.id);		
+		windowVO = GridWindowVO.get(SystemIDs.WINDOW_TEST, 1);
+		fields = windowVO.Tabs.get(0).getFields();
+		optional = fields.stream().filter(e -> e.AD_Field_ID == testRecordIdFieldId).findFirst();
+		assertTrue(optional.isPresent(), "Not Found, Record ID field of Test Window");
+		
+		assertTrue(gridWindowVOCache.getHit() > gridWindowVOCacheHit, "GridWindowVO cache hit doesn't increase as expected");
+		assertTrue(gridTabVOsCache.getHit() > gridTabVOsCacheHit, "GridTabVOs cache hit doesn't increase as expected");		
+		gridWindowVOCacheHit = gridWindowVOCache.getHit();
+		gridTabVOsCacheHit = gridTabVOsCache.getHit();
+		
+		//test GridWindowVO.get call with not advanced role
+		Env.setContext(Env.getCtx(), Env.AD_ROLE_ID, DictionaryIDs.AD_Role.GARDEN_WORLD_ADMIN_NOT_ADVANCED.id);		
+		windowVO = GridWindowVO.get(SystemIDs.WINDOW_TEST, 1);
+		fields = windowVO.Tabs.get(0).getFields();
+		optional = fields.stream().filter(e -> e.AD_Field_ID == testRecordIdFieldId).findFirst();
+		assertFalse(optional.isPresent(), "Record ID field of Test Window found for not advanced role");
+		
+		assertTrue(gridWindowVOCache.getHit() > gridWindowVOCacheHit, "GridWindowVO cache hit doesn't increase as expected");
+		assertTrue(gridTabVOsCache.getHit() > gridTabVOsCacheHit, "GridTabVOs cache hit doesn't increase as expected");		
+		gridWindowVOCacheHit = gridWindowVOCache.getHit();
+		gridTabVOsCacheHit = gridTabVOsCache.getHit();
+		
+		//test session cache with advanced role
+		Env.setContext(Env.getCtx(), Env.AD_ROLE_ID, DictionaryIDs.AD_Role.GARDEN_WORLD_ADMIN.id);
+		Env.setContext(Env.getCtx(), Env.AD_SESSION_ID, System.currentTimeMillis()+"");
```

Comment:
Hi @hengsin 

This test is throwing a lot of errors related to:
```
12:38:12.589===========> Env.getContextAsInt: (#AD_Session_ID) = 1711453092377 [1]
java.lang.NumberFormatException: For input string: "1711453092377"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:668)
	at java.base/java.lang.Integer.parseInt(Integer.java:786)
	at org.compiere.util.Env.getContextAsInt(Env.java:735)
	at org.compiere.model.MSession.get(MSession.java:73)
	at org.compiere.model.PO.delete(PO.java:4143)
	at org.compiere.model.PO.deleteEx(PO.java:4308)
	at org.idempiere.test.adwindow.GridWindowTest.testUserDefWin(GridWindowTest.java:275)
```

I tried changing the assignment of #AD_Session_ID to `ThreadLocalRandom.current().nextInt(10000000,Integer.MAX_VALUE)` instead of `System.currentTimeMillis()+""`

And then the error changed to:
```
12:35:54.952===========> MSession.loadPO: NO Data found for AD_Session_ID=1641036372 [1]
java.lang.Exception
	at org.compiere.model.PO.loadPO(PO.java:1591)
	at org.compiere.model.PO.load(PO.java:1506)
	at org.compiere.model.PO.load(PO.java:1457)
	at org.compiere.model.PO.<init>(PO.java:249)
	at org.compiere.model.PO.<init>(PO.java:162)
	at org.compiere.model.X_AD_Session.<init>(X_AD_Session.java:40)
	at org.compiere.model.MSession.<init>(MSession.java:161)
	at org.compiere.model.MSession.get(MSession.java:78)
	at org.compiere.model.PO.delete(PO.java:4143)
	at org.compiere.model.PO.deleteEx(PO.java:4308)
	at org.idempiere.test.adwindow.GridWindowTest.testUserDefWin(GridWindowTest.java:276)
```

Is it OK that the test class throws those errors?

Reply:
Miss that. No, should fix the error since it should be a pretty straightforward one.

Reply:
Fix pushed.